package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosDBLoadBalancingPolicy.java

```diff
@@ -0,0 +1,359 @@
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.azure.cosmos.cassandra;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+import com.datastax.driver.core.querybuilder.BuiltStatement;
+import com.google.common.collect.AbstractIterator;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * CosmosLoadBalancingPolicy allows you to specify readDC and writeDC to route read and write requests to their
+ * corresponding datacenters.
+ * globalEndpoint can also be used for writeDC. This allows the client to gracefully failover
+ * when the default write region is changed. dnsExpirationInSeconds is essentially the max duration to figure out the
+ * failover scenario. By default, it is 60 seconds.
+ * If readDC is specified, we prioritize nodes in the readDC for read requests.
+ * If writeDC is specified, we prioritize nodes in the writeDC for write requests.
+ */
+public class CosmosDBLoadBalancingPolicy implements LoadBalancingPolicy {
+
+    public static CosmosDBLoadBalancingPolicy buildFrom(Builder builder) {
+        validate(builder);
+        return new CosmosDBLoadBalancingPolicy(builder.readDC, builder.writeDC, builder.globalEndpoint, builder.dnsExpirationInSeconds);
+    }
+
+    /**
+     * Initializes the list of hosts in read, write, local, and remote categories.
+     */
+    @Override
+    public void init(Cluster cluster, Collection<Host> hosts) {
+        CopyOnWriteArrayList<Host> readLocalDCAddresses = new CopyOnWriteArrayList<Host>();
+        CopyOnWriteArrayList<Host> writeLocalDCAddresses = new CopyOnWriteArrayList<Host>();
+        CopyOnWriteArrayList<Host> remoteDCAddresses = new CopyOnWriteArrayList<Host>();
+
+        List<InetAddress> dnsLookupAddresses = new ArrayList<InetAddress>();
+        if (!globalContactPoint.isEmpty()) {
+            dnsLookupAddresses = Arrays.asList(getLocalAddresses());
+        }
+
+        for (Host host : hosts) {
+            if (!this.readDC.isEmpty() && host.getDatacenter().equals(readDC)) {
```

Comment:
Do we need this check?

Reply:
this.readDC.isEmpty()?

Reply:
just a sanity check. If the host's data center field is populated as "", it still shouldn't work.